### DISCUSSION TITLE
Error en MVN

### DISCUSSION THREAD

**Discussion by ciromartin** [dds-jv/foro#247] [Consultas de herramientas] [answered]
Hola buenas tardes, cambie de notebook por el trabajo y haciendo el TP Individual del filesystem me esta dando este error en el `mvn install`, en la notebook que tuve antes no tenia este problema

cuando corro los test desde el IDE van bien
![image](https://user-images.githubusercontent.com/12116895/119921048-b8dae780-bf43-11eb-8719-f59a1b2bf381.png)

cuando corro `mvn install`
![image](https://user-images.githubusercontent.com/12116895/119921235-06efeb00-bf44-11eb-81ae-07d1993c815c.png)

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.106 s
[INFO] Finished at: 2021-05-27T23:18:59-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project FileSystem: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd "/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin" && /usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/bin/java '-javaagent:/Users/a309711/.m2/repository/org/jacoco/org.jacoco.agent/0.8.2/org.jacoco.agent-0.8.2-runtime.jar=destfile=/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/jacoco.exec' -jar '/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/surefire/surefirebooter4412077909467508148.jar' '/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/surefire' 2021-05-27T23-18-59_407-jvmRun1 surefire15112604781232962355tmp surefire_010220717615565240606tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd "/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin" && /usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/bin/java '-javaagent:/Users/a309711/.m2/repository/org/jacoco/org.jacoco.agent/0.8.2/org.jacoco.agent-0.8.2-runtime.jar=destfile=/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/jacoco.exec' -jar '/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/surefire/surefirebooter4412077909467508148.jar' '/Users/a309711/Documents/utn/DDS/TPS Individuales/tpi2-sistema-de-archivos-ciromartin/target/surefire' 2021-05-27T23-18-59_407-jvmRun1 surefire15112604781232962355tmp surefire_010220717615565240606tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:669)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:282)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1183)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1011)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:857)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by nanderson87** [comment, marked as answer]
Hola @ciromartin 

1. El cambio de notebook representó un cambio sistema operativo o versión del mismo? En caso afirmativo cuál?
2. Si ejecutas un `mvn clean:install` te pasa lo mismo?

**Reply by ciromartin** [reply]
Hola @nanderson87 

1. Si cambie de notebook directamente, me dieron una mac. Antes tenia una HP con windows.
2. Los corro por separado el mvn clean, y mvn install con el mismo resultado. 

Porque cuando corro mvn clean:install me devuelve lo siguiente

a309711@MBRU309711 tpi2-sistema-de-archivos-ciromartin % mvn clean:install
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.149 s
[INFO] Finished at: 2021-05-28T10:03:08-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Could not find goal 'install' in plugin org.apache.maven.plugins:maven-clean-plugin:2.5 among available goals clean, help -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoNotFoundException

**Reply by nanderson87** [reply]
Ok. Me parece que el error en un principio puede venir por la versión de java que esta tratando de usar maven ya que en un punto se ve algo del estilo:

![image](https://user-images.githubusercontent.com/11875266/120052760-3f9fcb00-bffd-11eb-87a3-0b213a625e7b.png)


Verifica si tenes bien configurado la variable de  `JAVA_HOME`. Esta discusión te puede servir, ya que se habla de eso, en algunos casos para Windows pero hay casos de mac con la versión BigSur. Ya que al parecer en esa versión podía traer algunos problemas de cómo resuelve el path.
https://github.com/dds-jv/foro/discussions/180

**Reply by ciromartin** [reply]
Perfecto reconfigure las variables de entorno JAVA_HOME y M2_HOME en el archivo .zshrc y me funciono bien. Lo habia instalado con brew install mvn, lo desinstale, y instale a mano configurando la 2da variable. Muchas gracias !!

**Reply by nanderson87** [reply]
Buenisimo!
